File: app/src/main/java/com/example/market/io/server/HttpRequestManager.java
package com.example.market.io.server;


import com.example.market.util.Constant;

import java.io.IOException;
import java.io.OutputStream;
import java.net.HttpURLConnection;
import java.net.URL;
import java.nio.charset.StandardCharsets;


public class HttpRequestManager {

    public static HttpURLConnection executeRequest(String apiUrl, String requestMethod, String postData) {
        HttpURLConnection connection = null;

        try {
            URL ulr = new URL(apiUrl);
            connection = (HttpURLConnection) ulr.openConnection();
            connection.setRequestMethod(requestMethod);
            connection.setUseCaches(false);

            switch (requestMethod) {
                case Constant.RequestMethod.PUT:
                case Constant.RequestMethod.POST:
                    connection.setRequestProperty("Content-Type", "application/json");
                    connection.setDoInput(true);
                    connection.setDoOutput(true);
                    connection.connect();
                    OutputStream outputStream = connection.getOutputStream();
                    outputStream.write(postData != null ? postData.getBytes(StandardCharsets.UTF_8) : new byte[]{0});
                    outputStream.flush();
                    outputStream.close();
                    break;

                case Constant.RequestMethod.GET:
                    connection.connect();
                    break;
            }

        } catch (IOException e) {
            e.printStackTrace();
        }
        return connection;
    }
}

File: app/src/main/java/com/example/market/io/HttpRequestManager.kt
package com.example.market.io

import com.example.market.util.Constant
import java.io.IOException
import java.net.HttpURLConnection
import java.net.URL

object HttpRequestManager {
    fun executeRequest(
        apiUrl: String?,
        requestMethod: String?,
        postData: String?
    ): HttpURLConnection? {
        var connection: HttpURLConnection? = null
        try {
            val ulr = URL(apiUrl)
            connection = ulr.openConnection() as HttpURLConnection
            connection.requestMethod = requestMethod
            connection!!.useCaches = false
            when (requestMethod) {
                Constant.RequestMethod.PUT, Constant.RequestMethod.POST -> {
                    connection.setRequestProperty("Content-Type", "application/json")
                    connection.doInput = true
                    connection.doOutput = true
                    connection.connect()
                    val outputStream = connection.outputStream
                    outputStream.write(
                        if (postData != null)
                            postData.toByteArray() else byteArrayOf(
                            0
                        )
                    )
                    outputStream.flush()
                    outputStream.close()
                }
                Constant.RequestMethod.GET -> connection.connect()
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return connection
    }
}